src/main/java/org/commcare/formplayer/sqlitedb/ApplicationDBPath.java

```diff
@@ -11,17 +11,23 @@ class ApplicationDBPath extends DBPath {
     private String username;
     private String asUsername;
     private String appId;
+    private String appVersion;
 
-    ApplicationDBPath(String domain, String username, String asUsername, String appId) {
+    ApplicationDBPath(String domain, String username, String asUsername, String appId, String appVersion) {
         this.domain = domain;
         this.username = username;
         this.asUsername = asUsername;
         this.appId = appId;
+        this.appVersion = appVersion;
     }
 
     @Override
     public String getDatabasePath() {
-        return DbUtils.getDbPathForUser(domain, username, asUsername) + File.separator + appId;
+        String path = DbUtils.getDbPathForUser(domain, username, asUsername) + File.separator + appId;
+        if (appVersion != null) {
+            path = path + "_" + appVersion;
```

Comment:
optional: Seems like we are doing this concatenation of `appId +  "_" + appVersion` at multiple places in the PR and think would be nice to have this as a util method somewhere and reuse it.